ADAL/src/ADAuthenticationContext.m

```diff
@@ -346,6 +346,21 @@ - (void)acquireTokenWithResource:(NSString*)resource
     [request acquireToken:completionBlock];
 }
 
+#if TARGET_OS_WATCH
+- (void)acquireTokenSlientWithAuthData:(NSData *)authData
+                              resource:(NSString *)resource
+                              clientId:(NSString *)clientId
+                           redirectUri:(NSURL *)redirectUri
+                       completionBlock:(ADAuthenticationCallback)completionBlock
+{
+    API_ENTRY;
+    REQUEST_WITH_REDIRECT_URL(redirectUri, clientId, resource);
+    
+    [request setSilent:YES];
+    [request acquireTokenWithAuthData:authData completionBlock:completionBlock];
```

Comment:
Thanks for the clarification. The reason we are doing this work is trying to make sure a watch app can work *standalone* as much as possible. This means, once the watch get the initial authentication info from Phone, it should be able to refresh by itself without the help of the phone. That's why we are trying to pass the refresh token from phone to watch.